src/lib/y2storage/planned/partitions_distribution.rb

```diff
@@ -348,16 +348,18 @@ def too_many_primary_with_extended?(primary_spaces, ptable)
       def too_many_primary?(spaces, ptable)
         primary_spaces = spaces.select { |s| s.partition_type == :primary }
 
-        if !ptable.extended_possible?
+        if ptable.type.is?(:implicit)
+          num_partitions(primary_spaces) > ptable.max_primary
```

Comment:
We should never get here. Isn't it?

Anyway, I would compare `num_partitions(spaces) > ptable.max_primary`

Reply:
Why not?

Ok, better to use `spaces`.